src/core/manager.c

```diff
@@ -881,8 +881,16 @@ int manager_new(UnitFileScope scope, ManagerTestRunFlags test_run_flags, Manager
                         return r;
         }
 
-        if (MANAGER_IS_SYSTEM(m) && test_run_flags == 0) {
-                r = mkdir_label("/run/systemd/units", 0755);
+        if (test_run_flags == 0) {
+                if (MANAGER_IS_SYSTEM(m))
+                        r = mkdir_label("/run/systemd/units", 0755);
+                else {
+                        _cleanup_free_ char *units_path = NULL;
+                        if ((r = xdg_user_runtime_dir(&units_path, "/systemd/units")) < 0)
```

Comment:
hmm, can you do this in two steps? We are not fan of such compressed coding style... i.e. call the function in one line, and check the result in another...